holidays/long_weekends_finder.py

```diff
@@ -0,0 +1,98 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+import calendar
+from datetime import timedelta
+
+import holidays
+
+
+def find_long_weekends(country, year, month=None, language=None, exact_range=False):
+    # Validate year
+    if not isinstance(year, int) or not (1900 <= year <= 2100):
+        raise ValueError("Invalid year provided. Year must be an integer between 1900 and 2100.")
+
+    # Validate country (ensure it's a valid country code supported by the holidays library)
+    try:
+        holiday_obj = holidays.country_holidays(country, years=year, language=language)
+    except Exception as e:
+        raise ValueError(f"Error fetching holidays: {e}")
+
+    # Validate month (if provided)
+    if month is not None and (month < 1 or month > 12):
+        raise ValueError("Invalid month provided. Month must be between 1 and 12.")
+
+    weekends = holiday_obj.weekend
+    working_weekend_days = getattr(holiday_obj, "weekend_workdays", set())
+
+    holidays_filtered = {
+        d: name for d, name in holiday_obj.items() if month is None or d.month == month
+    }
```

Comment:
If we don't consider the `exact_range` value here, we may lose some dates. As example: UA, April 2019.

Also it would be useful, if `month` is not specified and `exact_range==False`, to include holidays from end of previous/beginning of next year. This is very useful in such a common task as planning New Year's vacations.

Reply:
I see no improvement for "UA, April 2019".

And "holidays from end of previous/beginning of next year" != "all holidays from previous and next years"